pkg/server/bootstrap_server.go

```diff
@@ -0,0 +1,122 @@
+package server
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	ignv2_2types "github.com/coreos/ignition/config/v2_2/types"
+	"github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ensure bootstrapServer implements the
+// Server interface.
+var _ = Server(&bootstrapServer{})
+
+type bootstrapServer struct {
+
+	// serverFileSystem is the file system used
+	// for reading files on the server.
+	serverFileSystem string
+
+	// serverBaseDir is the root, relative to which
+	// the machine pool, configs will be picked
+	serverBaseDir string
+
+	// serverKubeConfPath is the path on the local machine from
+	// where the kubeconfig should be read.
+	serverKubeConfPath string
+
+	// machineFileSystem defines the filesystem to be used
+	// for storing the files created by the server.
+	// eg: kubeconfig, node-annotations.
+	machineFileSystem string
+}
+
+// NewBootstrapServer initializes a new Bootstrap server that implements
+// the Server interface.
+func NewBootstrapServer(sfs, dir, mfs, kubeConf string) Server {
+	return &bootstrapServer{
+		serverFileSystem:   sfs,
+		serverBaseDir:      dir,
+		machineFileSystem:  mfs,
+		serverKubeConfPath: kubeConf,
+	}
+}
+
+// GetConfig fetches the machine config(type - Ignition) from the bootstrap server,
+// based on the pool request.
+// It returns nil for conf, error if the config isn't found. It returns a formatted
+// error if any other error is encountered during its operations.
+//
+// The method does the following:
+//
+// 1. Read the machine config pool by using the following path template:
+// 		"/<serverFileSystem>/<serverBaseDir>/<machineConfiPoolName>.yaml"
+//
+// 2. Read the currentConfig field from the Status and read the config file
```

Comment:
The ignition server should user the machineconfig specified in the machineconfigpool's .status.currentMachineConfig.
That means this dir structure is not required. just read the <machineconfig-name.yaml> ?